native/core/src/execution/datafusion/planner.rs

```diff
@@ -1347,12 +1349,24 @@ impl PhysicalPlanner {
 
                 match datatype {
                     DataType::Decimal128(_, _) => {
-                        Ok(Arc::new(SumDecimal::new("sum", child, datatype)))
+                        let func = AggregateUDF::new_from_impl(SumDecimal::new(
+                            "sum",
+                            Arc::clone(&child),
```

Comment:
Just for me to understand, what would happen if we do not do `Arc::clone()` here?

Reply:
We need to clone because we reference child again in the next statement. If I remove the close, the code fails to compile:

```
error[E0382]: use of moved value: `child`
    --> core/src/execution/datafusion/planner.rs:1357:72
     |
1347 |                 let child = self.create_expr(expr.child.as_ref().unwrap(), Arc::clone(&schema))?;
     |                     ----- move occurs because `child` has type `Arc<dyn datafusion_physical_expr::PhysicalExpr>`, which does not implement the `Copy` trait
...
1354 |                             child,
     |                             ----- value moved here
...
1357 |                         AggregateExprBuilder::new(Arc::new(func), vec![child])
     |                                                                        ^^^^^ value used here after move
```

Reply:
Hmm, I think the second parameter is `Arc<dyn PhysicalExpr>`. If it is not changed, it should be `child`?

Reply:
Oh, I see.

Reply:
Yes, we recently started using `Arc::clone(foo)` instead of `foo.clone()` to make it easy to see when we are just cloning an Arc (cheap) vs a more expensive clone operation. There is a clippy lint that checks that we are using this style.